test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(inputCommand.equals("add") || inputCommand.equals("delete") || inputCommand.equals("clear")) {
```

Comment:
This is not quite useful, cause it could not actually verify only above commands are mutating. I suggest adding an "else" here for non-mutating commands, which can then be checked whether the file was changed or not (e.g. updated time of file)